flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskUpdatePackages.java

```diff
@@ -205,8 +206,18 @@ private boolean updatePackageJsonDependencies(JsonObject packageJson,
             Map<String, String> applicationDevDependencies) throws IOException {
         int added = 0;
 
+        List<String> pinnedPlatformDependencies = new ArrayList<>();
+        final JsonObject platformPinnedDependencies = getPlatformPinnedDependencies();
+        Collections.addAll(pinnedPlatformDependencies,
+                platformPinnedDependencies.keys());
+
         // Add application dependencies
         for (Entry<String, String> dep : applicationDependencies.entrySet()) {
+            if (options.isReactEnabled() && isReactModuleAvailable()
+                    && !FrontendUtils.isPackageKeyReactComponents(dep.getKey())
+                    && pinnedPlatformDependencies.contains(dep.getKey())) {
+                continue;
+            }
```

Comment:
Trusting the `platformPinnedDependencies` content as `boolean reactComponents = pinnedPlatformDependencies.contains("@vaadin/react-components");` would make this:
```suggestion
            if (reactComponents 
                    && pinnedPlatformDependencies.contains(dep.getKey())) {
                continue;
            }
```

Also is this no here just to not add the `@NpmPackage` annotations from server Component definitions?

Reply:
Yes, basically so. Without this just having vaadin-core dependency would end up adding all web-components. Or having any dependency with `@NpmPackage` with core components.